go/runtime/txpool/txpool.go

```diff
@@ -231,38 +234,46 @@ func (t *txPool) SubmitTxNoWait(ctx context.Context, tx []byte, meta *Transactio
 }
 
 func (t *txPool) submitTx(ctx context.Context, rawTx []byte, meta *TransactionMeta, notifyCh chan *protocol.CheckTxResult) error {
-	tx := newTransaction(rawTx, txStatusPendingCheck)
+	// todo: in the new design, we'll be submitting to the check queue first, which won't have "first seen time"
+	// metadata. seems like it could still work out, as checking is single threaded, so we can still preserve the
+	// original order when we finish checking and offer to the main queue
```

Comment:
Sure if we are careful about ordering everywhere, including when checking a batch fails and batch is re-queued (e.g. `retryBatch` in the current check queue). Why can't we track first seen time?

Reply:
would have to be tracked in the externally-facing tx structure, which isn't settled yet

Reply:
moving timestamp to TxQueueMeta. also taking this opportunity to give it a descriptive field name, `FirstSeen`